sql/core/src/test/scala/org/apache/spark/sql/SubquerySuite.scala

```diff
@@ -204,6 +204,154 @@ class SubquerySuite extends QueryTest with SharedSparkSession {
     }
   }
 
+  test("SPARK-29145: JOIN Condition use QueryList") {
```

Comment:
Can we move it to SQLQueryTestSuite? 

It sounds like it does not contain any test case that check the EXISTS subquery? Could you also add it?

Reply:
> Can we move it to SQLQueryTestSuite?
> 
> It sounds like it does not contain any test case that check the EXISTS subquery? Could you also add it?

Ok, will raise a pr follow your comment.